src/ServiceControl/ExternalIntegrations/RemoveVersionInformationBehavior.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace ServiceControl.ExternalIntegrations
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.Pipeline;
+
+    public class RemoveVersionInformationBehavior : Behavior<IOutgoingPhysicalMessageContext>
+    {
+        public override Task Invoke(IOutgoingPhysicalMessageContext context, Func<Task> next)
+        {
+
+            if (context.Headers.TryGetValue(EnclosedTypesHeaderName, out var qualifiedTypeName)
+                && qualifiedTypeName.EndsWith(ContractsAssemblyIdentifier, StringComparison.OrdinalIgnoreCase))
```

Comment:
Wouldn't a type check based on assembly match be more efficient to a string compare? Especially because the string is quite lengthy.

Reply:
We don't have access to the type in this behavior. The header is added in the connector between logical and physical outgoing stage. At the same point the type information is lost.